de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/FuncElectricalDiagramTest.java

```diff
@@ -0,0 +1,203 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.swtbot.test;
+
+
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import org.eclipse.swtbot.eclipse.gef.finder.widgets.SWTBotGefConnectionEditPart;
+import org.eclipse.swtbot.eclipse.gef.finder.widgets.SWTBotGefEditPart;
+import org.eclipse.swtbot.eclipse.gef.finder.widgets.SWTBotGefEditor;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.concept.unittest.util.ConceptXmiLoader;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.Interface;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.InterfaceEnd;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.InterfaceType;
+import de.dlr.sc.virsat.model.extension.funcelectrical.model.InterfaceTypeCollection;
+import de.dlr.sc.virsat.model.extension.ps.model.ConfigurationTree;
+import de.dlr.sc.virsat.model.extension.ps.model.ElementConfiguration;
+
+
+
+/**
+ * 
+ * SWTBot test class for testing Funcelectrical Diagrams
+ *
+ */
+public class FuncElectricalDiagramTest extends ASwtBotTestCase {
+	
+	private SWTBotTreeItem repositoryNavigatorItem;
+	private SWTBotTreeItem configurationTree;
+	private SWTBotTreeItem elementConfiguration;
+	private SWTBotGefEditor diagramEditor;
+	
+	private Concept conceptFuncElectrical;
+	
+	private static final int DRAG1_COORDINATES_X = 100;
+	private static final int DRAG1_COORDINATES_Y = 100;
+	
+	private static final int DRAG2_COORDINATES_X = 100;
+	private static final int DRAG2_COORDINATES_Y = 200;
+	
+	private static final String ELEMENT_CONFIGURATION_NAME = "ElementConfiguration";
+
+	@Before
+	public void before() throws Exception {
+		super.before();
+		// create the necessary items for the test
```

Comment:
Well I think the comment descibes what is happening in a test setup in general. Maybe explain what the setup is doing, e.g. creating a blank diagram on a SEI or similar.